sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/ExpressionParserSuite.scala

```diff
@@ -804,14 +804,11 @@ class ExpressionParserSuite extends AnalysisTest {
       Literal(BigDecimal("90912830918230182310293801923652346786").underlying()))
     assertEqual("123.0E-28BD", Literal(BigDecimal("123.0E-28").underlying()))
     assertEqual("123.08BD", Literal(BigDecimal("123.08").underlying()))
+    val e = intercept[SparkArithmeticException](defaultParser.parseExpression("1.20E-38BD"))
     checkError(
-      exception = parseException("1.20E-38BD"),
-      errorClass = "_LEGACY_ERROR_TEMP_0061",
```

Comment:
Sorry, I didn't get. The PR title and description are telling only about one error class `LEGACY_ERROR_TEMP_1229` but your PR affects another one. Could you elaborate this and revert unrelated changes, or make PR's title/description consistent with PR's changes.

Reply:
Yeah, you're right.
I should have fixed `sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala` rather than tests.
Just reverted unrelated changes and fix them in proper way.
Thanks!